doc/source/examples_source/00-basic/07-volume_render.py

```diff
@@ -0,0 +1,199 @@
+# Load a dataset
+# --------------
+# Load Shuttle Session file included in the EnSight installation and render
+#
+# .. image:: /_static/07_volume_render_0.png
+
+###############################################################################
+# Load a dataset
+# --------------
+# Load Shuttle data included in the EnSight installation and render
+#
+# .. image:: /_static/03_ptrace_0.png
+
+xyz_file = f"{session.cei_home}/ensight{session.cei_suffix}/data/plot3d/shuttle.xyz"
+q_file = f"{session.cei_home}/ensight{session.cei_suffix}/data/plot3d/shuttle.q"
+session.load_data(
+    data_file=xyz_file,
+    result_file=q_file,
+    file_format="PLOT3D",
+    representation="3D_feature_2D_full",
+)
+session.show("image", width=800, height=600)
+
+###############################################################################
+# The PLOT3D reader only reads the volume by default. Now, extract a
+#  particular IJK range for the surface of the shuttle
+# ------------------------------------------------------------------
+
+sesse.data_partbuild.begin()
+sesse.case.select("Case 1")
+sesse.data_partbuild.data_type("structured")
+sesse.data_partbuild.group("OFF")
+sesse.data_partbuild.select_begin(1)
+sesse.data_partbuild.domain("all")
+sesse.data_partbuild.noderange_i(1, 53)
+sesse.data_partbuild.noderange_j(1, 63)
+sesse.data_partbuild.noderange_k(1, 1)
+sesse.data_partbuild.nodestep(1, 1, 1)
+sesse.data_partbuild.nodedelta(0, 0, 0)
+sesse.data_partbuild.description("Shuttle")
+sesse.data_partbuild.create()
+sesse.part.select_byname_begin("(CASE:Case 1)Shuttle")
+sesse.case.select("Case 1")
+sesse.data_partbuild.end()
+session.show("image", width=800, height=600)
+
+##############################################################################
+# Volume Render Flow
+# ------------------------------------------------------------------
+# 
+# .. image:: /_static/07_volume_render_1.png
+#
+#
+# Make the shuttle surface an uninteresting, uniform color grey
+#
+sesse.part.select_begin(2)
+sesse.part.colorby_palette("none")
+sesse.part.colorby_rgb(0.6,0.6,0.6)
+sesse.view.highlight_parts("OFF")
+#
+# Color the 3D flow field by the variable that we
+#  wish to visualize using Volume Rendering, Mach
+#
+sesse.part.select_begin(1)
+sesse.part.modify_begin()
+sesse.part.colorby_palette("Mach")
+sesse.legend.select_palette_begin("Mach")
+sesse.part.modify_end()
+sesse.legend.select_palette_begin("Mach")
+sesse.function.palette("Mach")
+sesse.function.palette("Mach")
+sesse.function.modify_begin()
+sesse.function.restore_predefinedpal("use_new_levels","Magma")
+sesse.function.modify_end()
+#
+#  Volume Rendering, in EnSight, is accomplished by subdividing
+#   the flow field into a finite number of hex elements and then
+#   doing the rendering of the variable in each box according to
+#   the normalized variable value and the opacity ('alpha') for
+#   ranges of values set using function points and the magnitude
+#
+sesse.tools.box("ON")
+sesse.tools.box("OFF")
```

Comment:
Pedantic.... why is there a tools box on, and then immediately a tools box off?

Reply:
fixed; removed them.